napari/plugins/_plugin_manager.py

```diff
@@ -84,6 +88,27 @@ def register(
             self.events.registered(value=name)
         return name
 
+    def unregister(
+        self,
+        name_or_object: Any,
+    ) -> Optional[Any]:
+        plugin = super().unregister(name_or_object)
+
+        # remove widgets, sample data
+
+        if type(name_or_object) is str:
+
+            for _dict in (
+                self._dock_widgets,
+                self._sample_data,
+                self._function_widgets,
+            ):
+                _dict.pop(name_or_object, None)
+
+        self.events.unregistered(value=name_or_object)
```

Comment:
This may add a little more code before the `super().unregister` call above, but we really ought to normalize the type being emitted by the `unregistered` event (it should always be the string name of the plugin)... so that callbacks don't have to check it.  here, I believe that `name_or_object` can be either a string, or a namespace (like a module).  So can you normalize that before emitting?  something like this should work, but you'll need to do it before the call to super()

```python
if isinstance(name_or_object, str):
    _name = name_or_object
else:
    _name = self.get_name(name_or_object)
```